share/wrf_tsin.F

```diff
@@ -46,17 +49,35 @@ SUBROUTINE wrf_tsin ( grid , ierr )
    
              IF (istatus == 0) THEN
    
-                ! Ignore first three lines, which constitute a header
-                READ(UNIT=iunit, FMT='(1X)')
+                ! The header will be parsed to see if lat/lon or i/j is given...
                 READ(UNIT=iunit, FMT='(1X)')
+                READ(UNIT=iunit, FMT='(32X,A7,2X,A7)') JorLat, IorLon
                 READ(UNIT=iunit, FMT='(1X)')
+                grid%tslist_ij = .false.
+                ! If the header contains "LAT" and "LON" then it will expect
+                !... lat/lon coords in the file
+                ! If the header has "J" and "I", then i,j will be used.
+                if(index(JorLat,'LAT').ne.0 .and. index(IorLon,'LON').ne.0) then
+                    grid%tslist_ij = .false.
+                elseif(index(JorLat,'J').ne.0 .and. index(IorLon,'I').ne.0) then
+                    grid%tslist_ij = .true.
+                endif
+
+
+
    
                 ! Read in time series locations
                 istatus = 0
                 DO WHILE (istatus == 0)
-                   READ(UNIT=iunit, FMT='(A25,1X,A5,1X,F7.3,1X,F8.3)', IOSTAT=istatus)            &
-                        grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1), &
-                        grid%lattsloc(grid%ntsloc+1), grid%lontsloc(grid%ntsloc+1)
+                   IF (config_flags%map_proj == 0 .OR. grid%tslist_ij) THEN !ideal run will only use i,j
+                       READ(UNIT=iunit, FMT='(A25,1X,A5,1X,I9,1X,I10)', IOSTAT=istatus)            &
+                       grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1), &
+                       grid%itsloc(grid%ntsloc+1), grid%jtsloc(grid%ntsloc+1)
```

Comment:
Is this correct? The header suggests that the coordinates are given as (J,I), but here we're reading them as (I,J). I'd actually prefer to give the coordinates as I first, then J, as that seems most natural; but that's only my opinion.

Reply:
I will run some tests to verify this - I kept it as (J,I) so it would be equivalent to lat/lon, but I agree that (I,J) is more natural. I will change it to (I,J) if this is not the case.

Reply:
I have verified that it is, indeed, reading in (I,J) but the header says (J,I). I have modified this so that the header must say (I,J); if it says (J,I) then no towers are generated. 

Also, I think it may be redundant for the 'IF' statement to include the `config_flags%map_proj == 0` statement now. Any idealized case will use (I,J) in the header, as well as, any real case that specifies (I,J). If everybody agrees, the new 'IF' statement will read as: `IF ( grid%tslist_ij) THEN`.